event-sourcing/event-repository/event-repository-jdbc/src/main/java/uk/gov/justice/services/eventsourcing/repository/jdbc/eventlog/EventLogJdbcRepository.java

```diff
@@ -112,20 +104,30 @@ public void insert(final EventLog eventLog) throws InvalidSequenceIdException {
      * @return a stream of {@link EventLog}. Never returns null.
      */
     public Stream<EventLog> findByStreamIdFromSequenceIdOrderBySequenceIdAsc(final UUID streamId, final Long versionFrom) {
-        List<EventLog> events;
 
-        try (final Connection connection = getDataSource().getConnection();
-             final PreparedStatement ps = connection.prepareStatement(SQL_FIND_BY_STREAM_ID_AND_SEQUENCE_ID)) {
+        try {
+            final PreparedStatementWrapper ps = preparedStatementWrapperOf(SQL_FIND_BY_STREAM_ID_AND_SEQUENCE_ID);
 
             ps.setObject(1, streamId);
             ps.setLong(2, versionFrom);
 
-            events = extractResults(ps);
-        } catch (SQLException | NamingException e) {
+            return streamOf(ps);
+        } catch (SQLException e) {
             throw new JdbcRepositoryException(format(READING_STREAM_EXCEPTION, streamId), e);
         }
+    }
 
-        return events.stream();
+    /**
+     * Returns a Stream of {@link EventLog}
+     *
+     * @return a stream of {@link EventLog}. Never returns null.
+     */
+    public Stream<EventLog> findAll() {
+        try {
+            return streamOf(preparedStatementWrapperOf(SQL_FIND_ALL));
```

Comment:
nope, because we are dealing with streams here. the resulting stream needs to be in try with resources when it's consumed